### PR TITLE
functionality : PeerAddressType for Private Peering and ignore case logic

### PR DESCRIPTION
**Pull request by kilobyte97**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ public override void Execute()
                 peering.SharedKey = this.SharedKey;
             }
 
-            if (this.PeerAddressType == IPv6)
+            if (string.Equals(this.PeerAddressType, "IPv6", StringComparison.OrdinalIgnoreCase))
```
Comment (wyunchi-ms) [inline src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs:72]: As the logic of checking `PeerAddressType` is very common. I suggest to create some util function for these logic.

Comment (kilobyte97) [reply src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs:72]: Follow up logic beyond the check is different in different files. Not sure what to do with utility function.

Are you suggesting to define a bool util function which implements
(string.Equals(this.PeerAddressType, "IPv6", StringComparison.OrdinalIgnoreCase))

**Comment by kilobyte97** [comment]
guys can you review ?

File: src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs (hunk 1 above)
Comment (wyunchi-ms) [reply src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs:72]: Yes. Also for `IPv4`.

**Comment by wyunchi-ms** [comment]
Hi @kilobyte97 , please take a look at the comment.

File: src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs (hunk 1 above)
Comment (kilobyte97) [reply src/Network/Network/ExpressRouteCircuit/Peering/AddAzureExpressRouteCircuitPeeringConfigCommand.cs]: done.

**Comment by kilobyte97** [comment]
resolved comment.
Please take a look

**Comment by kilobyte97** [comment]
Thanks @wyunchi-ms for the file updates.
any action item before this gets checked in ?

**Comment by kilobyte97** [comment]
@number213 @wyunchi-ms  please review.

**Pull request merged**